tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -26,9 +26,107 @@
 )
 from tardis.montecarlo.montecarlo_numba import njit_dict
 from numba.typed import List
+from tardis.util.base import is_notebook
+from IPython.display import display
+import tqdm
 
+if is_notebook():
+    pbar = tqdm.notebook.tqdm
+else:
+    pbar = tqdm.tqdm
 
-def montecarlo_radial1d(model, plasma, runner):
+packet_pbar = pbar(
+    dynamic_ncols=True,
+    bar_format="{bar}{percentage:3.0f}% of packets propagated, iteration 0/?",
+)
+if type(packet_pbar).__name__ == "tqdm_notebook":
+    packet_pbar.container.close()
+
+
+def update_packet_pbar(i, current_iteration, total_iterations, total_packets):
```

Comment:
I'm not sure if this function should be in the base montecarlo numba file.

Reply:
Will adding this function to `tardis/util/base.py` be a good idea?